File: Java/Programs-based-on-Numbers/30-decimal-to-octal-conversion.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter decimal number:");
        int decimal = getInput.nextInt();

        String octal = "";

        while (decimal > 0) {
            int remainder = decimal%8;
            octal = remainder + octal;
            decimal /= 8;
        }
        
        System.out.println("Octal value is: " + octal);
        getInput.close();
    }
}

// Using built-in method
// String octal = Integer.toOctalString(decimal);

File: Python/2-Programs-based-on-Numbers/30-decimal-to-octal-conversion.py
decimal = int(input("Enter Decimal number: "))
octal = ""

if decimal == 0:
    print("Octal value is: 0")

else:
    while decimal > 0:
        remainder = decimal%8
        octal = str(remainder) + octal
        decimal = decimal//8
    
    print("Octal value is:", octal)


# Using built-in method
# octal = oct(decimal)